packages/core/src/session.ts

```diff
@@ -187,7 +187,7 @@ class Session {
     const result = this._run(validatedQuery, params, async connection => {
       const bookmarks = await this._bookmarks()
       this._assertSessionIsOpen()
-      return (connection as Connection).protocol().run(validatedQuery, params, {
+      return (connection as Connection).run(validatedQuery, params, {
```

Comment:
(Somewhat unrelated:) I tried to understand why this cast is necessary. 
I found the type definition `type ConnectionConsumer = (connection: Connection | null) => any | undefined | Promise<any> | Promise<undefined>` and wondered why connection could be `null` in that type but here in the callback it is assumed/asserted it's not.

Reply:
The connection can be null if `ConnectionHolder` is not initialised with success.